project-api/ballerina-projects/src/main/java/io/ballerina/projects/balo/BaloFiles.java

```diff
@@ -0,0 +1,200 @@
+            throw new RuntimeException("'organization' does not exists in 'package.json'");
+        }
+        if (packageJson.getName() == null || "".equals(packageJson.getName())) {
+            throw new RuntimeException("'name' does not exists in 'package.json'");
+        }
+        if (packageJson.getVersion() == null || "".equals(packageJson.getVersion())) {
+            throw new RuntimeException("'version' does not exists in 'package.json'");
+        }
+    }
+
+    private static ModuleData loadModule(Path modulePath) {
+        // check module path exists
+        if (!Files.exists(modulePath)) {
+            throw new RuntimeException("module does not exists:" + modulePath);
+        }
+
+        String moduleName = String.valueOf(modulePath.getFileName());
+        if (moduleName.contains(".")) { // not default module
+            moduleName = moduleName.split("\\.")[1];
+            moduleName = moduleName.replace("/", "");
+        }
+
+        // validate moduleName
+        if (!ProjectUtils.validateModuleName(moduleName)) {
+            throw new RuntimeException("Invalid module name : '" + moduleName + "' :\n" +
+                    "Module name can only contain alphanumerics, underscores and periods " +
+                    "and the maximum length is 256 characters");
+        }
+        List<DocumentData> srcDocs = loadDocuments(modulePath);
+        List<DocumentData> testSrcDocs = Collections.emptyList();
+
+        // TODO Read Module.md file. Do we need to? Balo creator may need to package Module.md
+        return ModuleData.from(modulePath, srcDocs, testSrcDocs);
+    }
+
+    private static List<DocumentData> loadDocuments(Path dirPath) {
+        try (Stream<Path> pathStream = Files.walk(dirPath, 1)) {
+            return pathStream
+                    .filter(matcher::matches)
+                    .map(BaloFiles::loadDocument)
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static DocumentData loadDocument(Path documentFilePath) {
+        Path fileNamePath = documentFilePath.getFileName();
+        // IMO, fileNamePath cannot be null in this case.
+        String name = fileNamePath != null ? fileNamePath.toString() : "";
+        return DocumentData.from(name, String.valueOf(documentFilePath));
+    }
+
+    private static List<ModuleData> loadOtherModules(Path modulesDirPath, Path defaultModulePath) {
+        if (!Files.isDirectory(modulesDirPath)) {
+            throw new RuntimeException("'modules' directory does not exists:" + modulesDirPath);
+        }
+
+        try (Stream<Path> pathStream = Files.walk(modulesDirPath, 1)) {
+            return pathStream
+                    .filter(path -> !path.equals(modulesDirPath))
+                    .filter(path -> !String.valueOf(path).equals("/" + defaultModulePath + "/"))
```

Comment:
Should we use `File.separator` instead of `"/"`?

Reply:
Added